lib/Dialect/ModArith/Conversions/ModArithToArith/ModArithToArith.cpp

```diff
@@ -250,6 +269,127 @@ struct ConvertMac : public OpConversionPattern<MacOp> {
   }
 };
 
+struct ConvertModSwitch : public OpConversionPattern<ModSwitchOp> {
+  ConvertModSwitch(mlir::MLIRContext *context)
+      : OpConversionPattern<ModSwitchOp>(context) {}
+
+  using OpConversionPattern::OpConversionPattern;
+
+  auto getInteger(ImplicitLocOpBuilder *b, IntegerType integerType) const {
+    return [=](const APInt &value) mutable {
+      return b->create<arith::ConstantOp>(IntegerAttr::get(integerType, value));
+    };
+  }
```

Comment:
This can be simplified (perhaps enough to inline) via `b.create<arith::ConstantIntOp>(integerType, value)`